### PR TITLE
Update rabbitmq-components.mk to correctly reference components

### PR DESCRIPTION
**Pull request by HoloRin**
This adds a way to fetch sub-components from the monorepo and rabbitmq-components.mk has been updated accordingly.

Should address #13

### CODE REVIEW

**Comment by lukebakken** [comment]
Testing this out today.

File: rabbitmq-components.mk [hunk 1]
```diff
@@ -37,73 +37,73 @@ endif
...
+dep_rabbitmq_web_dispatch             = git_rmq-subfolder rabbitmq-web-dispatch $(current_rmq_ref) $(base_rmq_ref) master
+dep_rabbitmq_web_stomp                = git_rmq-subfolder rabbitmq-web-stomp $(current_rmq_ref) $(base_rmq_ref) master
+dep_rabbitmq_web_stomp_examples       = git_rmq-subfolder rabbitmq-web-stomp-examples $(current_rmq_ref) $(base_rmq_ref) master
+dep_rabbitmq_web_mqtt                 = git_rmq-subfolder rabbitmq-web-mqtt $(current_rmq_ref) $(base_rmq_ref) master
+dep_rabbitmq_web_mqtt_examples        = git_rmq-subfolder rabbitmq-web-mqtt-examples $(current_rmq_ref) $(base_rmq_ref) master
 dep_rabbitmq_website                  = git_rmq rabbitmq-website $(current_rmq_ref) $(base_rmq_ref) live master
 dep_toke                              = git_rmq toke $(current_rmq_ref) $(base_rmq_ref) master
 
-dep_rabbitmq_public_umbrella          = git_rmq rabbitmq-public-umbrella $(current_rmq_ref) $(base_rmq_ref) master
+# dep_rabbitmq_public_umbrella          = git_rmq rabbitmq-public-umbrella $(current_rmq_ref) $(base_rmq_ref) master
```
Comment (lukebakken) [inline rabbitmq-components.mk]: Should this have been removed entirely?

**Review by lukebakken — COMMENTED** [review]
This is an interesting strategy for a fix. Is it documented? Also this will cause this `rabbitmq-components.mk` to diverge from the canonical one we have here -

https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbit_common/mk/rabbitmq-components.mk

Is that what we want for plugins? cc @dumbbell

File: rabbitmq-components.mk (hunk 1 above)
Comment (HoloRin) [reply rabbitmq-components.mk]: Possibly yes. It's not clear to me what purposes the umbrella serves when it is a dep. I suspect I commented it out to ensure it was not needed, and forgot to delete it. On the other hand, it doesn't hurt anything, and the umbrella is still the most convenient way to test 3.7.x versions of the broker (if the plugin author was concerned about compatibility).

**Comment by HoloRin** [comment]
The "canonical" version in the monorepo is mostly unused now, as Erlang.mk no longer uses those lines to fetch those deps. It would probably be safe to update that version as well, though I have wondered if they should in fact be removed, as their presence may be misleading.

The `git-subfolder` fetch method for deps is mentioned in https://erlang.mk/guide/deps.html#_fetch_methods - `git_rmq-subfolder` attempts to follow the pattern, but is hard-coded to refer to the monorepo.

There is still plenty more Makefile cleanup that could happen in the monorepo, but I have been reluctant to remove too many things before our patterns are settled.

**Comment by noxdafox** [comment]
Hello,

this interests me as well as currently I cannot build my plugins due to the monorepo changes.

Personally, I'd love the `rabbitmq-components.mk` file to be stored in a unique place so I can rely on it rather than trying to maintain a copy myself.

Previously, I used to fetch it at build time as shown [here](https://github.com/noxdafox/rabbitmq-message-deduplication/blob/master/rabbitmq-components.mk). This allowed me to trust I always used the latest version which kept me worry-free for quite a long time.

**Comment by HoloRin** [comment]
Hi @noxdafox , I recently opened https://github.com/rabbitmq/rabbitmq-server/pull/2918 which results in a single `rabbitmq-components.mk` at the root of rabbitmq-server. Once it is merged you should be able to update your Makefile to fetch it from the new location.

**Comment by noxdafox** [comment]
Thanks @pjk25! I will test it as soon as I can.

**Comment by HoloRin** [comment]
@noxdafox I made an additional update to rabbitmq-components.mk this morning that should make things a bit more efficient, so I'd recommend trying the latest version.

**Pull request merged**